Retain purchase originalJson, signature, userID for use in server-side receipt verification

fadookie opened the pull request:
I want to be able to do my own server-side receipt verification; this requires the originalJson and signature fields from the IabPurchase object. This patch captures this data and makes it available in the MarketPurchaseEvent object. I submitted another pull request to get this data back to Unity here: https://github.com/soomla/unity3d-store/pull/422

refaelos commented:
This pull-request is great but it breaks implementation of other packages. We'll merge it in a few days (after investigation)

fadookie commented:
Great! I don't have a build pipeline set up yet for anything other than the android store so I didn't test against other packages, sorry. I guess maybe the fix here is to just make these extras optional in the constructor for MarketPurchaseEvent?

Thanks, and shabbat shalom. :)

fadookie commented:
I'm not entirely sure if this fixes the issue you were having, but I fixed the constructors in MarketPurchaseEvent() to restore the old signatures.

I also added support for passing back the Amazon User ID, also for purposes of server-side receipt verification. This is used in a new pull request: https://github.com/soomla/android-store-amazon/pull/3
